sources/logdog/src/log_request.rs

```diff
@@ -205,12 +218,114 @@ where
     Ok(())
 }
 
+/// Copies all file matching the glob pattern given by `request.instructions` to the tempdir with filename and path
+/// same as source file.
+fn handle_glob_request<P>(request: &LogRequest<'_>, tempdir: P) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    let glob_paths = glob(request.instructions).context(error::ParseGlobPattern {
+        pattern: request.instructions.to_string(),
+    })?;
+    for entry in glob_paths {
+        if let Ok(path) = entry {
+            if path.is_dir() {
+                // iterate the directory and copy files in the directory, nested
+                // sub-directories files will be copied only if matched by glob pattern.
+                for candidate in WalkDir::new(&path)
+                    .max_depth(1)
+                    .into_iter()
+                    .flat_map(|e| e.context(error::DirectoryWalk))
+                    .map(|e| e.into_path())
+                {
+                    if candidate.is_file() {
+                        file_copy(&request, &tempdir, &candidate)?;
+                    }
+                }
+            } else {
+                file_copy(&request, &tempdir, &path)?;
+            }
+        }
+    }
+    Ok(())
+}
+
+fn file_copy<P>(request: &LogRequest<'_>, destdir: P, src_filepath: &PathBuf) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    // with glob pattern there are chances of multiple targets with same name, therefore
+    // we maintain source file path and name in destination directory.
+    // Eg. src file path "/a/b/file" will be converted to "dest_dir/a/b/file"
```

Comment:
It's too bad that I didn't do this with all the files we already collect, but this behavior makes sense. 

Unfortunately it might be a little confusing when you unpack that tarball and some things are at the top and other things replicate their on-host paths. One possibility would be to put all of these paths into a dir at the top of the tarball like `/glob`, but I don't think that's any better.